classes/CMSCategory.php

```diff
@@ -498,23 +498,37 @@ public static function checkBeforeMove($id_cms_category, $id_parent)
         }
     }
 
-    public static function getLinkRewrite($id_cms_category, $id_lang)
+    /**
+     * Get the rewrite link of the given CMS Category.
+     *
+     * @param int $idCMSCategory CMS Category ID
+     * @param int $idLang Language ID
+     * @param int $idShop Shop id
+     *
+     * @return bool|mixed
+     */
+
+    public static function getLinkRewrite($idCMSCategory, $idLang, $idShop = null)
     {
-        if (!Validate::isUnsignedId($id_cms_category) || !Validate::isUnsignedId($id_lang)) {
+        if (!Validate::isUnsignedId($idCMSCategory) || !Validate::isUnsignedId($idLang)) {
             return false;
         }
 
-        if (isset(self::$_links[$id_cms_category . '-' . $id_lang])) {
-            return self::$_links[$id_cms_category . '-' . $id_lang];
+        $idShop = $idShop ?: isset(Context::getContext()->shop->id)
+        ? (int) Context::getContext()->shop->id
+        : (int) Configuration::get('PS_SHOP_DEFAULT');
+
+        if (isset(self::$_links[$idCMSCategory . '-' . $idLang . '-' . $idShop])) {
+            return self::$_links[$idCMSCategory . '-' . $idLang . '-' . $idShop];
         }
 
         $result = Db::getInstance()->getRow('
 		SELECT cl.`link_rewrite`
 		FROM `' . _DB_PREFIX_ . 'cms_category` c
 		LEFT JOIN `' . _DB_PREFIX_ . 'cms_category_lang` cl ON c.`id_cms_category` = cl.`id_cms_category`
-		WHERE `id_lang` = ' . (int) $id_lang . '
-		AND c.`id_cms_category` = ' . (int) $id_cms_category);
-        self::$_links[$id_cms_category . '-' . $id_lang] = $result['link_rewrite'];
+		WHERE `id_lang` = ' . (int) $idLang . '
+		AND c.`id_cms_category` = ' . (int) $idCMSCategory);
```

Comment:
Yes actualy the PR is WIP because i have this issue :

Deprecated: Unparenthesized `a ?: b ? c : d` is deprecated. Use either `(a ?: b) ? c : d` or `a ?: (b ? c : d)` 

I need to ask in which branch the PR will go and also the minimum PHP version of the dev if it's 9.1

Reply:
The use of nested ternaries is really not recommended, it is a source of problems and not very fluid.
I gave you a simple solution above ;)

Reply:
And be careful because Validate::isUnsignedId() does not prevent zeros.